liberty-maven-plugin/src/main/java/io/openliberty/tools/maven/server/DevMojo.java

```diff
@@ -349,57 +463,64 @@ public boolean recompileBuildFile(File buildFile, List<String> artifactPaths, Th
-                            dependencyIds.clear();
-                        }
-
-                        // update dependencies
-                        for (Artifact artifact : updatedArtifacts) {
-                            for (Dependency dependency : dependencies) {
-                                if (artifact.getArtifactId().equals(dependency.getArtifactId())) {
-                                    this.existingDependencies.add(dependency);
-                                    break;
-                                }
-                            }
-                        }
-                        if (!isUsingBoost() && unhandledChange) {
-                            log.warn(
-                                    "Unhandled change detected in pom.xml. Restart liberty:dev mode for it to take effect.");
-                        } else {
-                            this.existingPom = modifiedPom;
-                        }
-                        return true;
-                    } else if (isUsingBoost()) {
-                        this.existingPom = modifiedPom;
-                        return true;
-                    } else if (unhandledChange) {
-                        log.warn(
-                                "Unhandled change detected in pom.xml. Restart liberty:dev mode for it to take effect.");
+                if (restartServer) {
+                    // TODO: restart server automatically
+                    // - stop Server
+                    // - create server or runBootMojo
+                    // - install feature
+                    // - deploy app
+                    // - start server
+                    log.error("Changes for Liberty server bootstrap properties or environment variables or JVM options "
+                            + "in the pom.xml has been detected, Restart liberty:dev mode for it to take effect.");
+                    project = backupProject;
+                    session.setCurrentProject(backupProject);
+                    return false;
+                } else {
+                    if (isUsingBoost() && (createServer || runBoostPackage)) {
+                        log.info("Running boost:package");
+                        runBoostMojo("package", false);
+                    } else if (createServer) {
+                        runLibertyMojoCreate();
+                    } else if (redeployApp) {
+                        runLibertyMojoDeploy();
+                    }
+                    if (installFeature) {
+                        runLibertyMojoInstallFeature(null);
                     }
                 }
-            } catch (Exception e) {
-                log.debug("Could not recompile pom.xml", e);
+                if (!(restartServer || createServer || redeployApp || installFeature || runBoostPackage)) {
+                    // pom.xml is changed but not affecting dev mode
+                    log.debug("changes in the pom.xml are not monitored by dev mode");
```

Comment:
Should this also print some sort of `info` message?

Reply:
If we print an info message, what do we expect the user to do? 
I even think we should update the project in the current session with the latest updated project, and return true to run the project integration tests.  Maybe the changes are for the test plugin (e.g. additional system properties)

Reply:
> we should update the project in the current session with the latest updated project, and return true to run the project integration tests.

I agree.

Reply:
ok